fleetspeak/src/server/dbtesting/clientstore_suite.go

```diff
@@ -286,13 +286,48 @@ func clientStoreTest(t *testing.T, ds db.Store) {
 	if err != nil {
 		t.Fatalf("Unexpected error when writing client resource-usage data: %v", err)
 	}
-	records, err := ds.FetchResourceUsageRecords(ctx, clientID, 100)
+	startTimeRange := &tpb.Timestamp{
+		Seconds: 82,
+		Nanos:   4,
+	}
+	endTimeRange := &tpb.Timestamp{
+		Seconds: 84,
+		Nanos:   4,
+	}
+	records, err := ds.FetchResourceUsageRecords(ctx, clientID, startTimeRange, endTimeRange)
 	if err != nil {
 		t.Errorf("Unexpected error when trying to fetch resource-usage data for client: %v", err)
 	}
 	if len(records) != 1 {
 		t.Fatalf("Unexpected number of records returned. Want %d, got %v", 1, len(records))
 	}
+	startNoRecordsTimeRange := &tpb.Timestamp{
```

Comment:
2 things:
* The test method is gigantic. While we're at it, can we refactor the FetchResourceUsageRecords check in a separate test method? See ListClientsTest as an example (https://github.com/google/fleetspeak/blob/8062d1823419a8527d19b5cbb4b8b5a3e7fb9a33/fleetspeak/src/server/dbtesting/clientstore_suite.go#L448). I'd suggest introducing a FetchResourceUsageRecordsTest.
* I'd suggest converting the FetchResourceUsage testing logic to be table-driven. See, for example: https://github.com/google/fleetspeak/blob/8062d1823419a8527d19b5cbb4b8b5a3e7fb9a33/fleetspeak/src/server/dbtesting/clientstore_suite.go#L96